bootstrap/src/basis/array.ml

```diff
@@ -209,18 +205,19 @@ module Seq = struct
           end in
           let elm, t' = T.next t in
           let a = Stdlib.Array.make l elm in
```

Comment:
Kinda weird that `Stdlib.Array.make` initializes every single element to `elm` before we make a second pass and correct it all. Would it make more sense to use `Stdlib.Array.init`?

Reply:
Yeah, this kinda sucks.  The problem we have here is that `Seq` is a pure external iterator, and the only way we can use it with `Stdlib.Array.init` is to lexically bind a mutable variable.  I have a different mechanism in mind for the Hemlock version (https://github.com/BranchTaken/Hemlock/blob/master/src/basis/array.hl#L137)

Reply:
Incidentally, the whole reason we pervasively use `Seq` is to avoid double initialization, across a variety of data types.  Absent this rough spot in the bootstrap library, I think `Seq` is turning out to serve that purpose well.